pmacApp/src/pmacController.cpp

```diff
@@ -2157,15 +2157,24 @@ asynStatus pmacController::writeFloat64(asynUser *pasynUser, epicsFloat64 value)
     }
 
 
-  } else if (function == PMAC_C_MotorRes_){
+  } 
+  else if (function == PMAC_C_MotorRes_){
     pAxis->setResolution(value);
     // Direct resolution parameter will always match the raw motor
     setDoubleParam(pAxis->axisNo_, PMAC_C_DirectRes_, value);
-  } else if (function == PMAC_C_MotorOffset_){
+  } 
+  else if (function == PMAC_C_MotorOffset_){
     pAxis->setOffset(value);
     // Direct offset parameter will always match the raw motor
     setDoubleParam(pAxis->axisNo_, PMAC_C_DirectOffset_, value);
-  } else if (function == PMAC_C_DirectMove_){
+
+    int csNum = this->getAxis(pAxis->axisNo_)->getAxisCSNo();
+    if (csNum > 0) {
+        csResetAllDemands = true;
+        makeCSDemandsConsistent();
```

Comment:
Remove `makeCSDemandsConsistent` from here to keep  `csResetAllDemands` as true until the next axis move command.